### ISSUE TITLE
Automatic channel ID creation uses invalid character

### ISSUE THREAD

**Issue by greg-szabo** [cosmos/relayer#363]
I run the demo setup as described in the documentation, from the `master` branch (SDK v0.40.0).

After creating the link, I get `channel-0` as the channel ID for both networks.

After sending a transaction from ibc0 to ibc1 (relaying worked fine), the query command for the wallet on ibc1 fails:

```
$ rly q bal chain1
  timeout: 10s
panic: invalid denom: transfer/channel-0/uatom

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/types.NewCoin(0xc000e9b160, 0x18, 0xc000f99a80, 0xc000ee4585, 0x5, 0xc000e9b160)
	/Users/greg/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.40.0/types/coin.go:23 +0x8a
  rpc-addr: http://localhost:26657
github.com/cosmos/relayer/cmd.queryBalanceCmd.func1(0xc000f17b80, 0xc000f28170, 0x1, 0x1, 0x0, 0x0)
	/Users/greg/git/cosmos/relayer/cmd/query.go:257 +0x5bb
github.com/spf13/cobra.(*Command).execute(0xc000f17b80, 0xc000f28150, 0x1, 0x1, 0xc000f17b80, 0xc000f28150)
	/Users/greg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x5deba60, 0x0, 0x0, 0x0)
	/Users/greg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/greg/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/cosmos/relayer/cmd.Execute()
	/Users/greg/git/cosmos/relayer/cmd/root.go:107 +0x4d
main.main()
	/Users/greg/git/cosmos/relayer/main.go:21 +0x25
```

The `NewCoin()` function does a validation on the denom name and that fails possibly because the channel ID contains the `-` character which is considered invalid in a denom.

**Comment by cryptophonic** [comment]
Thanks for tracking this down - was trying to figure out a different issue this weekend and realized the demo was not working either.  Since the sdk.Coin array is being used only for output in this case, it's probably acceptable to skip the NewCoin() logic and just create the type directly:

```
diff --git a/cmd/query.go b/cmd/query.go
index bc819a5..f4115e7 100644
--- a/cmd/query.go
+++ b/cmd/query.go
@@ -254,7 +254,7 @@ $ %s q bal ibc-1 --ibc-denoms`, appName, appName, appName)),
                                                switch {
                                                case c.Amount.Equal(sdk.NewInt(0)):
                                                case c.Denom == d.IBCDenom():
-                                                       out = append(out, sdk.NewCoin(d.GetFullDenomPath(), c.Amount))
+                                                       out = append(out, sdk.Coin{ Denom: d.GetFullDenomPath(), Amount: c.Amount })
                                                default:
                                                        out = append(out, c)
                                                }
```

**Comment by colin-axner** [comment]
This has been fixed in both v0.7.0 and on the current master branch
